frontend/webpack.config.ts

```diff
@@ -120,6 +120,18 @@ const config: Configuration = {
   optimization: {
     splitChunks: {
       chunks: 'all',
+      cacheGroups: {
+        'monaco': {
+          test: /[\\/]node_modules[\\/](monaco-.*|yaml-language-server)[\\/]/,
```

Comment:
I believe this is causing the monaco chunk to be loaded on console load. Why is this needed?

Reply:
My intention was to give the biggest chunk a sensible name, i.e. `monaco-chunk` instead of `<number-may-vary>-chunk`.

This chunk currently has two roots:
- `node_modules/monaco-editor`
- `node_modules/yaml-language-server`

The slight growth of chunk size (2.59 MB => 2.78 MB) is due to above `test` having

```
monaco-.* | yaml-language-server
```

instead of more specific

```
monaco-editor | yaml-language-server
```

If we don't want _all_ Monaco related dependencies grouped in this chunk, I can change that.

As for the chunk loaded on Console startup - thanks for noticing this, I've missed that. It seems that some other code (depending on roots in `monaco-chunk`) which is part of initial chunk triggers this.

I think this can be fixed with `chunks: 'all'` => `chunks: 'async'` ([docs](https://webpack.js.org/plugins/split-chunks-plugin/#splitchunkschunks)).

Reply:
I believe this is caused by

```
WARNING in ./node_modules/monaco-editor-core/esm/vs/editor/common/services/editorSimpleWorker.js 374:12-383:17
Critical dependency: the request of a dependency is an expression
```

This means `monaco-editor-core` is included in the main bundle. Since we've added this regex to`cacheGroups`, it puts `monaco-editor-core` with the rest of the monaco editor dependencies and loads *everything* on console load.  This more than doubles the vendor JS we load.

Can we simply remove this from `cacheGroups`? We should be really careful about the changes we make there.

I think it makes sense for `kubevirt-web-ui-components` and might eliminate the need for putting all the kubevirt pages in one chunk. But I'm OK either way

Reply:
> I believe this is caused by
> 
> ```
> WARNING in ./node_modules/monaco-editor-core/esm/vs/editor/common/services/editorSimpleWorker.js 374:12-383:17
> Critical dependency: the request of a dependency is an expression
> ```

Yes, this is a problem with `monaco-editor` library, referring to code which does a dynamic `require` call (module name isn't a string literal) and therefore hinders webpack optimization.

> Can we simply remove this from `cacheGroups`? We should be really careful about the changes we make there.

Agreed, will do.

> I think it makes sense for `kubevirt-web-ui-components` and might eliminate the need for putting all the kubevirt pages in one chunk. But I'm OK either way

I'll retest this to be sure.